### PR TITLE
auth with google

### PR DESCRIPTION
**Pull request by ashgolan**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -17,7 +20,19 @@ app.use(express.json());
...
-app.use(cors());
+app.use(
+	cors({
+		origin: 'http://localhost:3000',
+		methods: 'GET,POST,PUT,DELETE',
+		credentials: true,
+	})
+);
+app.use(
+	cookieSession({ name: 'session', keys: ['lama'], maxAge: 24 * 60 * 60 * 100 })
```
Comment (AmeerLalaDeveloper) [inline index.js:31]: change keys to movieApp

Comment (pini85) [inline index.js:31]: the cookie keys should come from a env. Make it a long unique one

File: passport.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
```
Comment (pini85) [inline passport.js:1]: passport.js should be under services

File: passport.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+// const FACEBOOK_APP_ID = 'your id';
+// const FACEBOOK_APP_SECRET = 'your id';
+
+passport.use(
+	new GoogleStrategy(
+		{
+			clientID: GOOGLE_CLIENT_ID,
+			clientSecret: GOOGLE_CLIENT_SECRET,
+			callbackURL: '/auth/google/movie-app',
```
Comment (pini85) [inline passport.js:20]: best practice to add the callbackURL with callback. eg : "/auth/google/callback"

File: passport.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		},
+		function (accessToken, refreshToken, profile, done) {
+			done(null, profile);
```
Comment (pini85) [inline passport.js:23]: I would like to see in this PR also logic to save the user in the DB if does not exists. And retrieve an existing user.

File: passport.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+// 		}
+// 	)
+// );
+
+passport.serializeUser((user, done) => {
+	done(null, user);
+});
+
+passport.deserializeUser((user, done) => {
+	done(null, user);
```
Comment (pini85) [inline passport.js:59]: deserialize users from our db

File: routes/auth.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	});
+});
+
+authRoute.get('/logout', (req, res) => {
+	req.logout();
+	res.redirect(CLIENT_URL);
+});
+
+authRoute.get(
+	'/google',
```
Comment (pini85) [inline routes/auth.js:29]: please use as the path auth/google

File: routes/auth.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import express from 'express';
+import passport from 'passport';
+export const authRoute = express.Router();
+const CLIENT_URL = 'http://localhost:3000/';
```
Comment (pini85) [inline routes/auth.js:4]: this is only working for local. How about in production? Also put them in our env files.

**Review by pini85 — CHANGES_REQUESTED** [review]

File: passport.js (hunk 3 above)
Comment (ashgolan) [reply passport.js:20]: i dont understand what callback should we create

File: passport.js (hunk 5 above)
Comment (ashgolan) [reply passport.js:59]: ??

File: routes/auth.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+	'/google',
+	passport.authenticate('google', { scope: ['profile'] })
+);
+
+authRoute.get(
+	'/google/callback',
+	passport.authenticate('google', {
+		successRedirect: CLIENT_URL,
+		failureRedirect: '/login/failed',
+	})
```
[the remark this answers is not in the document]
Comment (ashgolan) [reply routes/auth.js:38]: ??

File: passport.js (hunk 2 above)
Comment (ashgolan) [reply passport.js]: done

File: passport.js (hunk 4 above)
Comment (ashgolan) [reply passport.js]: done

**Pull request closed without merging**